.github/actions/get-jtreg/action.yml

```diff
@@ -0,0 +1,72 @@
+#
+# Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+#
+# This code is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 only, as
+# published by the Free Software Foundation.  Oracle designates this
+# particular file as subject to the "Classpath" exception as provided
+# by Oracle in the LICENSE file that accompanied this code.
+#
+# This code is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+# version 2 for more details (a copy is included in the LICENSE file that
+# accompanied this code).
+#
+# You should have received a copy of the GNU General Public License version
+# 2 along with this work; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+# or visit www.oracle.com if you need additional information or have any
+# questions.
+#
+
+name: 'Get JTReg'
+description: 'Download JTReg from cache or source location'
+outputs:
+  path:
+    description: 'Path to the installed JTReg'
+    value: ${{ steps.path-name.outputs.path }}
+
+runs:
+  using: composite
+  steps:
+    - name: 'Get JTReg version configuration'
+      id: version
+      uses: ./.github/actions/config
+      with:
+        var: JTREG_VERSION
+
+    - name: 'Check cache for JTReg'
+      id: get-cached-jtreg
+      uses: actions/cache@v3
+      with:
+        path: jtreg/installed
+        key: jtreg-${{ steps.version.outputs.value }}
+
+    - name: 'Checkout the JTReg source'
+      uses: actions/checkout@v3
+      with:
+        repository: sormuras/jtreg
+        ref: 7903206-build-on-msys2
+        path: jtreg/src
+      if: steps.get-cached-jtreg.outputs.cache-hit != 'true'
```

Comment:
Just leaving the comment here so we don't forget to change it to mainline jtreg location before integrating this PR :)